bin/main.rb

```diff
@@ -1,76 +1,88 @@
 #!/usr/bin/env ruby
+require_relative '../lib/tictactoe.rb'
 
-# user instructions
+puts
 puts 'Welcome to **Tic-Tac-toe**'
-puts 'In this game whichever player gets the first similar letters in a row, column or diagonal Wins!.'
-puts "There are only two letters to pick from 'X' and 'O'."
-puts 'At the begining of the game, each player will choose which letter they will be playing with,'
-puts 'once one player has selected a letter, the otherone is assigned the other letter automatically.'
-puts "Note: Player names are optional (just press 'Enter' to continue)"
-puts ''
-puts ''
+Tictactoe.print_table_empty
+puts "In this game, there are only two letters to play with 'X' and 'O'. Whichever player gets the first similar letters in an entire row, column or diagonal WINS!!!."
+puts
+puts "At the begining of the game, player 1 will choose either 'X' or 'O' and the other letter will be assigned to player 2 automatically."
+puts
 
-# Get user input and set it to a variable
-p 'Enter name of player 1 or press Enter to skip'
+puts 'Enter name of player 1 or press Enter to skip'
 input1 = gets.chomp
-input1 == '' ? 'player 1' : input1
-p 'enter a letter to play with'
-player_1_letter = gets.chomp
-p "#{input1} chose #{player_1_letter}"
+input1 == '' ? input1 = 'player 1' : input1
+
+puts 'enter a letter to play with'
+player_1_letter = gets.chomp.upcase
+until %w[X O].include?(player_1_letter)
+  puts 'enter a valid letter to play with'
+  letter = gets.chomp.upcase
+  player_1_letter = letter if Tictactoe.valid_letter?(letter)
+end
+
+puts "#{input1} chose #{player_1_letter}"
 
-p 'Enter name of player 2 or press Enter to skip'
+puts 'Enter name of player 2 or press Enter to skip'
 input2 = gets.chomp
-input2 == '' ? 'player 2' : input2
+input2 == '' ? input2 = 'player 2' : input2
 player_2_letter = 'O' if player_1_letter == 'X'
 player_2_letter = 'X' if player_1_letter == 'O'
```

Comment:
Move that validation into the TicTacToe class.
The result will be returned and used here.